File: 김진효/week04/12982_예산.py
# [성능 요약] 메모리: 9.04 MB 시간: 0.01 ms

#  그리디 인트로인가 ^-^
#  정렬해주고 낮은 지원금부터 지원 ㄱㄱ
def solution(d, budget):
    answer = 0
    d.sort()
    for b in d:
        budget -= b

        if budget < 0:
            return answer

        answer += 1

    return answer


print(solution([1,3,2,5,4],9))
print(solution([2,2,3,3],10))

File: 백태균/week04/12982_예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {

        Arrays.sort(d);  // 작은 순으로 순회하면 가장 많은 경우의 수를 바로 알 수 있어서 오름차순으로 정렬

        int sum = 0;
        int cnt = 0;

        // 순회하면서
        for (int i=0; i<d.length; i++) {
            sum += d[i];  // 하나씩 sum에 더하기

            if (sum <= budget) {  // 여태까지 더한 값이 budget 보다 작거나 같다면
                cnt++;  // cnt 증가
            } else {  // 여태까지 더한 값이 budget 보다 크다면
                break;  // 이제 더 이상 찾을 필요가 없으니깐 정지
            }
        }

        return cnt;
    }
}